akka-persistence-typed/src/main/scala/akka/persistence/typed/Retention.scala

```diff
@@ -34,22 +35,26 @@ final case class RetentionCriteria(snapshotEveryNEvents: Long, keepNSnapshots: L
     // Delete old events, retain the latest
     math.max(0, lastSequenceNr - (keepNSnapshots * snapshotEveryNEvents))
   }
+
+  /** Java API. */
+  def withKeepNSnapshots(keepN: Long): RetentionCriteria =
+    copy(keepNSnapshots = keepN)
+
+  /** Java API. */
+  def withDeleteEventsOnSnapshot(): RetentionCriteria =
+    copy(deleteEventsOnSnapshot = true)
 }
 
 object RetentionCriteria {
 
-  def apply(): RetentionCriteria =
+  def default: RetentionCriteria =
     RetentionCriteria(snapshotEveryNEvents = 1000L, keepNSnapshots = 2L, deleteEventsOnSnapshot = false)
 
+  def snapshotEvery(numberOfEvents: Long): RetentionCriteria =
+    apply(numberOfEvents, 2L)
+
   /** Scala API. */
   def apply(snapshotEveryNEvents: Long, keepNSnapshots: Long): RetentionCriteria =
     RetentionCriteria(snapshotEveryNEvents, keepNSnapshots, deleteEventsOnSnapshot = false)
```

Comment:
Inconsistent Java and Scala APIs now, so maybe this should go as well?

Reply:
I think `snapshotEvery can be along the lines of this, because now there will be no default, only 0 and user is required to set both to use.
```scala
def snapshotEvery(n: Long, keepNSnapshots: Long): RetentionCriteria =
```